src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/RollupMapperService.kt

```diff
@@ -15,16 +15,272 @@
+                is RollupJobValidationResult.Invalid -> return sourceIndexMappingResult
+                is RollupJobValidationResult.Failure -> return sourceIndexMappingResult
+            }
+        }
+
+        return RollupJobValidationResult.Valid
+    }
+
+    @Suppress("ReturnCount", "ComplexMethod")
+    private suspend fun isSourceIndexMappingsValid(index: String, rollup: Rollup): RollupJobValidationResult {
+        try {
+            val res = when (val getMappingsResult = getMappings(index)) {
+                is GetMappingsResult.Success -> getMappingsResult.response
+                is GetMappingsResult.Failure ->
+                    return RollupJobValidationResult.Failure(getMappingsResult.message, getMappingsResult.cause)
+            }
+
+            val indexMapping: MappingMetadata = res.mappings[index][_DOC]
+            val indexMappingSource = indexMapping.sourceAsMap
+
+            val issues = mutableSetOf<String>()
+            // Validate source fields in dimensions
+            rollup.dimensions.forEach { dimension ->
+                if (!isFieldInMappings(dimension.sourceField, indexMappingSource))
+                    issues.add("missing field ${dimension.sourceField}")
+
+                when (dimension) {
+                    is DateHistogram -> {
+                        // TODO: Validate if field is date type: date, date_nanos?
+                    }
+                    is Histogram -> {
+                        // TODO: Validate field types for histograms
+                    }
+                    is Terms -> {
+                        // TODO: Validate field types for terms
+                    }
+                }
+            }
+
+            // Validate source fields in metrics
+            rollup.metrics.forEach { metric ->
+                if (!isFieldInMappings(metric.sourceField, indexMappingSource))
+                    issues.add("missing field ${metric.sourceField}")
+
+                // TODO: Validate field type for metrics,
+                //  are all Numeric field types valid?
+            }
+
+            return if (issues.isEmpty()) {
+                RollupJobValidationResult.Valid
+            } else {
+                RollupJobValidationResult.Invalid("Invalid mappings for index [$index] because $issues")
+            }
+        } catch (e: Exception) {
+            return RollupJobValidationResult.Failure("Failed to validate the source index mappings", e)
+        }
+    }
+
+    /**
+     * Checks to see if the given field name is in the mappings map.
+     *
+     * The field name can be a path in the format "field1.field2...fieldn" so each field
+     * will be checked in the map to get the next level until all subfields are checked for,
+     * in which case true is returned. If at any point any of the fields is not in the map, false is returned.
+     */
+    private fun isFieldInMappings(fieldName: String, mappings: Map<*, *>): Boolean {
```

Comment:
optional - TODO: fix the assumption that the field has always "properties"

Reply:
Fixed in PR: https://github.com/opendistro-for-elasticsearch/index-management/pull/343